.github/workflows/ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  HF_SCRIPTS_VERSION: main
+  HF_ALLOW_CODE_EVAL: 1
+
+jobs:
+
+  check_code_quality:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.6"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .[quality]
+      - name: Check quality
+        run: |
+          black --check --line-length 119 --target-version py36 tests src benchmarks datasets metrics
+          isort --check-only tests src benchmarks datasets metrics
+          flake8 tests src benchmarks datasets metrics
+
+  test:
+    needs: check_code_quality
```

Comment:
The `check_code_quality` job must complete successfully before the `test` jobs will run. 

This saves CI running time (billing).

Reply:
I think I would still run the tests even if the code quality check doesn't succeed. This will avoid having to review a first time to tell the user to fix code quality, and then later review a second time to fix the tests

Reply:
In the specific use case you mention, the reviewer should say to the contributor: you should run locally both the code quality checks and the tests